### ISSUE TITLE
ECTO: submission #10 is broken

### ISSUE THREAD

**Issue by alexskr** [ncbo/bioportal-project#327]
latest submission #10 for ECTO ontology is missing description and contacts, which are required properties.

**Comment by alexskr** [comment]
added missing properties and re-processed ontology.  
encountered errors with indexing

**Comment by alexskr** [comment]
errors:

```
E, [2024-10-08T13:37:42.034272 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 1 times..."]
E, [2024-10-08T13:37:44.034961 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 2 times..."]
E, [2024-10-08T13:37:46.035546 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 3 times..."]
E, [2024-10-08T13:37:48.036490 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 4 times..."]
E, [2024-10-08T13:37:50.037080 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 5 times..."]
E, [2024-10-08T13:37:52.037676 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 6 times..."]
E, [2024-10-08T13:37:54.038926 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 7 times..."]
E, [2024-10-08T13:37:56.039495 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 8 times..."]
E, [2024-10-08T13:37:58.040073 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 9 times..."]
E, [2024-10-08T13:38:00.040697 #323627] ERROR -- : ["Thread 1: Exception while mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264. Retrying 10 times..."]
E, [2024-10-08T13:38:02.041383 #323627] ERROR -- : ["Thread 1: Error mapping attributes for http://purl.obolibrary.org/obo/FOODON_00001264:"]
E, [2024-10-08T13:38:02.042719 #323627] ERROR -- : ["Thread 1: NoMethodError: undefined method `map' for #<RDF::Literal:0x1ceb98(\"http://purl.obolibrary.org/obo/FOODON_03411254\"^^<http://www.w3.org/2001/XMLSchema#anyURI>)>\nDid you mean?  tap after retrying 10 times...\n/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/goo-74a012eebb94/lib/goo/base/resource.rb:250:in `block in map_attributes'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/goo-74a012eebb94/lib/goo/base/resource.rb:222:in `each'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/goo-74a012eebb94/lib/goo/base/resource.rb:222:in `map_attributes'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:132:in `rescue in block (4 levels) in index'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:118:in `block (4 levels) in index'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `each'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `block (3 levels) in index'"]
E, [2024-10-08T13:38:02.044789 #323627] ERROR -- : ["\n\nGoo::Base::AttributeNotLoaded: Attribute `synonym` is not loaded for http://purl.obolibrary.org/obo/FOODON_00001264. Loaded attributes: #<Set: {:submission, :label, :prefLabel}>.\n"]
E, [2024-10-08T13:38:02.045705 #323627] ERROR -- : [["/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/goo-74a012eebb94/lib/goo/base/settings/settings.rb:300:in `block in shape_attribute'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/utils/ontology_csv_writer.rb:44:in `write_class'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:148:in `block (5 levels) in index'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/models/ontology_submission.rb:90:in `synchronize'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/models/ontology_submission.rb:90:in `synchronize'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:147:in `block (4 levels) in index'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `each'", "/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `block (3 levels) in index'"]]
E, [2024-10-08T13:38:02.045904 #323627] ERROR -- : ["Goo::Base::AttributeNotLoaded: Attribute `synonym` is not loaded for http://purl.obolibrary.org/obo/FOODON_00001264. Loaded attributes: #<Set: {:submission, :label, :prefLabel}>.\n/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/goo-74a012eebb94/lib/goo/base/settings/settings.rb:300:in `block in shape_attribute'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/utils/ontology_csv_writer.rb:44:in `write_class'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:148:in `block (5 levels) in index'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/models/ontology_submission.rb:90:in `synchronize'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/models/ontology_submission.rb:90:in `synchronize'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:147:in `block (4 levels) in index'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `each'\n\t/srv/ontoportal/ncbo_cron/vendor/bundle/ruby/3.0.0/bundler/gems/ontologies_linked_data-8572cd25722c/lib/ontologies_linked_data/services/submission_process/operations/submission_indexer.rb:117:in `block (3 levels) in index'"]
I, [2024-10-08T13:38:02.099597 #323627]  INFO -- : ["Indexing ontology properties: ECTO..."]
```

**Comment by mdorf** [comment]
This should be fixed in ncbo/ontologies_linked_data#220

**Comment by alexskr** [comment]
fixed
